CRM/Dedupe/BAO/RuleGroup.php

```diff
@@ -196,13 +196,13 @@ public function fillTable() {
     if ($this->params && !$this->noRules) {
       $tempTableQuery = "CREATE TEMPORARY TABLE dedupe (id1 int, weight int, UNIQUE UI_id1 (id1)) ENGINE=InnoDB";
       $insertClause = "INSERT INTO dedupe (id1, weight)";
-      $groupByClause = "GROUP BY id1";
+      $groupByClause = "GROUP BY id1, weight";
```

Comment:
@eileenmcnaughton i think given that we are grouping on one id i think it would be ok but not 100% sure